src/app/home/pages/ClustersPage/components/AddEditClusterModal/index.tsx

```diff
@@ -1,54 +1,62 @@
 import React, { useState, useEffect, useContext } from 'react';
 import { connect } from 'react-redux';
-import AddEditClusterForm from './AddEditClusterForm';
+import AddEditClusterForm, { IFormValues } from './AddEditClusterForm';
 import { Modal } from '@patternfly/react-core';
 import { ClusterActions } from '../../../../../cluster/duck/actions';
 import {
   defaultAddEditStatus,
   AddEditMode,
   createAddEditStatus,
   AddEditState,
+  IAddEditStatus,
 } from '../../../../../common/add_edit_state';
 import { PollingContext } from '../../../../duck/context';
 import { IReduxState } from '../../../../../../reducers';
+import { ICluster } from '../../../../../cluster/duck/types';
+import { IClusterInfo } from '../../helpers';
 
-// TODO add types here
-const AddEditClusterModal = ({
+interface IAddEditClusterModal {
+  addEditStatus: IAddEditStatus;
+  initialClusterValues: IClusterInfo;
+  isOpen: boolean;
+  isPolling: boolean;
+  checkConnection: (name) => void;
+  clusterList: ICluster[];
+  isAdmin: boolean;
+  addCluster: (cluster) => void;
+  cancelAddEditWatch: () => void;
+  onHandleClose: () => void;
+  resetAddEditState: () => void;
+  updateCluster: (updatedCluster) => void;
+}
+
+const AddEditClusterModal: React.FunctionComponent<IAddEditClusterModal> = ({
   addEditStatus,
   initialClusterValues,
   isOpen,
   isPolling,
   checkConnection,
   clusterList,
-  cluster,
   isAdmin,
-  ...props
-}) => {
+  addCluster,
+  cancelAddEditWatch,
+  onHandleClose,
+  resetAddEditState,
+  updateCluster,
+}: IAddEditClusterModal) => {
   if (!isAdmin) return null;
 
   const pollingContext = useContext(PollingContext);
-  const [currentClusterName, setCurrentClusterName] = useState(
```

Comment:
Removing this breaks the add edit cluster flow. There is an opportunity to refactor this but removing it leaves the add/edit/check connection in a broken state.

Reply:
Hey Gilles - submitted a PR to address some changes i ran into during my add/edit token work. Lines up nicely with what you have already started here. See PR https://github.com/gildub/mig-ui/pull/6